code/modules/mob/living/carbon/superior_animal/golem/golem.dm

```diff
@@ -86,3 +113,38 @@
 
 	// Poof
 	qdel(src)
+
+/mob/living/carbon/superior_animal/golem/gib(var/anim = icon_gib, var/do_gibs = FALSE)
+	. = ..(anim, FALSE)  // No gibs when gibbing a golem (no blood)
+
+/mob/living/carbon/superior_animal/golem/destroySurroundings()
+	// Get next turf the golem wants to walk on
+	var/turf/T = get_step_towards(src, target_mob)
+
+	if(iswall(T))  // Wall breaker attack
+		T.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)
+		return
+
+	for (var/obj/structure/window/obstacle in T)
+		obstacle.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)
+		return
+
+	for (var/obj/machinery/door/window/obstacle in T)
+		obstacle.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)
+		return
+
+	var/obj/structure/obstacle = locate(/obj/structure, T)
+	if (istype(obstacle, /obj/structure/window) || istype(obstacle, /obj/structure/closet) || istype(obstacle, /obj/structure/table) \
+		|| istype(obstacle, /obj/structure/grille) || istype(obstacle, /obj/structure/low_wall) || istype(obstacle, /obj/structure/railing) \
+		|| istype(obstacle, /obj/structure/girder))
+		obstacle.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)
```

Comment:
Maybe even `obstacle?.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)`.

Reply:
> Maybe even `obstacle?.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), attacktext, TRUE)`, since there is no `else` case.

You're right, your solution is better. I adapted `/mob/living/carbon/superior_animal/proc/destroySurroundings()` without thinking too much about it. I'll make the change.